### PR TITLE
ktls-utils: init at 0.10

### LINKED ISSUE #273827: Package request: ktls-utils
**Project description**

TLS handshake utilities for in-kernel TLS consumers. Notably, provides a critical userspace component for NFS over TLS.

**Metadata**

* homepage URL: https://github.com/oracle/ktls-utils
* source URL: https://github.com/oracle/ktls-utils
* license: GPL2
* platforms: Linux

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by arthsmn**
## Description of changes

Homepage: https://github.com/oracle/ktls-utils

Closes #273827

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/kt/ktls-utils/package.nix [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  src = fetchFromGitHub {
+    owner = "oracle";
+    repo = "ktls-utils";
+    rev = "ktls-utils-${version}";
+    hash = "sha256-1HceFajSSq1D3IMeGQF+i/iW1lL2opLLWFfsRHW9atg=";
+  };
+
+  nativeBuildInputs = [
+    automake
+    autoconf
```
Comment (NickCao) [inline pkgs/by-name/kt/ktls-utils/package.nix:27]:
```suggestion
    autoreconfHook
```

File: pkgs/by-name/kt/ktls-utils/package.nix [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    keyutils
+    glib
+    libnl
+  ];
+
+  outputs = [ "out" "man" ];
+
+  preConfigure = ''
+    ./autogen.sh
+  '';
```
Comment (NickCao) [inline pkgs/by-name/kt/ktls-utils/package.nix:42]:
```suggestion
```

File: pkgs/by-name/kt/ktls-utils/package.nix [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+, stdenv
+, fetchFromGitHub
+, automake
+, autoconf
+, pkg-config
+, gnutls
+, keyutils
+, glib
+, libnl
+, withSystemd ? stdenv.isLinux
```
Comment (NickCao) [inline pkgs/by-name/kt/ktls-utils/package.nix:11]:
```suggestion
, withSystemd ? lib.meta.availableOn stdenv.hostPlatform systemd
```

File: pkgs/by-name/kt/ktls-utils/package.nix [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+  preConfigure = ''
+    ./autogen.sh
+  '';
+
+  configureFlags = lib.optional withSystemd [ "--with-systemd" ];
+
+  makeFlags = lib.optional withSystemd [ "unitdir=$(out)/lib/systemd/system" ];
+
+  doCheck = !stdenv.isDarwin;
```
Comment (NickCao) [inline pkgs/by-name/kt/ktls-utils/package.nix:48]: Add comment on why.

File: pkgs/by-name/kt/ktls-utils/package.nix [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+  meta = with lib; {
+    description = "TLS handshake utilities for in-kernel TLS consumers";
+    homepage = "https://github.com/oracle/ktls-utils";
+    changelog = "https://github.com/oracle/ktls-utils/blob/${src.rev}/NEWS";
+    license = licenses.gpl2Only;
+    maintainers = with maintainers; [ ];
+    mainProgram = "ktls-utils";
+    platforms = platforms.all;
```
Comment (NickCao) [inline pkgs/by-name/kt/ktls-utils/package.nix:57]: Isn't ktls only available on Linux?

File: pkgs/by-name/kt/ktls-utils/package.nix (hunk 2 above)
Comment (arthsmn) [reply pkgs/by-name/kt/ktls-utils/package.nix:42]: Done.

File: pkgs/by-name/kt/ktls-utils/package.nix (hunk 1 above)
Comment (arthsmn) [reply pkgs/by-name/kt/ktls-utils/package.nix:27]: Done.

File: pkgs/by-name/kt/ktls-utils/package.nix (hunk 3 above)
Comment (arthsmn) [reply pkgs/by-name/kt/ktls-utils/package.nix:11]: Done.

File: pkgs/by-name/kt/ktls-utils/package.nix (hunk 4 above)
Comment (arthsmn) [reply pkgs/by-name/kt/ktls-utils/package.nix:48]: Not needed anymore.

File: pkgs/by-name/kt/ktls-utils/package.nix (hunk 5 above)
Comment (arthsmn) [reply pkgs/by-name/kt/ktls-utils/package.nix:57]: Sorry, saw some issues about FreeBSD but it does only support linux.

**Review by NickCao — APPROVED** [review]

**Pull request merged**